realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -0,0 +1,124 @@
+     *                                         an un-managed collection.
+     * @return A sorted view of the collection.
+     */
+    RealmResults<E> sort(String fieldName);
+
+    /**
+     * Sorts existing collection in-place.
+     *
+     * @param fieldName the field name to sort by. Only fields of type boolean, short, int, long, float, double, Date,
+     *                  and String are supported.
+     * @param sortOrder the direction to sort by.
+     * @throws java.lang.IllegalArgumentException if field name does not exist or has an invalid type.
+     * @throws java.lang.IllegalStateException if the Realm is closed, called on the wrong thread or the collection is
+     *                                         an un-managed collection.
+     * @return A sorted copy of the collection.
+     */
+    RealmResults<E> sort(String fieldName, Sort sortOrder);
+
+    /**
+     * Sorts existing {@link io.realm.RealmResults} in-place using two fields.
+     *
+     * @param fieldName1 first field name.
+     * @param sortOrder1 sort order for first field.
+     * @param fieldName2 second field name.
+     * @param sortOrder2 sort order for second field.
+     * @throws java.lang.IllegalArgumentException if a field name does not exist or has an invalid type.
+     * @throws java.lang.IllegalStateException if the Realm is closed, called on the wrong thread or the collection is
+     *                                         an un-managed collection.
+     * @return a sorted view of the collection.
+     */
+    RealmResults<E> sort(String fieldName1, Sort sortOrder1, String fieldName2, Sort sortOrder2);
+
+    /**
+     * Sorts existing {@link io.realm.RealmResults} in-place.
+     *
+     * @param fieldNames an array of field names to sort by. Only fields of type boolean, short, int, long, float,
+     *                   double, Date, and String are supported.
+     * @param sortOrders the directions to sort by.
+     * @throws java.lang.IllegalArgumentException if a field name does not exist or has an invalid type.
+     * @throws java.lang.IllegalStateException if the Realm is closed, called on the wrong thread or the collection is
+     *                                         an un-managed collection.
+     * @return a sorted view of the collection.
+     */
+    RealmResults<E> sort(String[] fieldNames, Sort[] sortOrders);
+
+    /**
+     * Deletes the object at the given index from the Realm. This also removes it from the collection.
+     *
+     * @param location the array index identifying the object to be removed.
+     * @throws IndexOutOfBoundsException if {@code location < 0 || location >= size()}.
+     * @throws java.lang.IllegalStateException if the Realm is closed, called on the wrong thread or the collection is
+     *                                         an un-managed collection.
+     */
+    void deleteFromRealm(int location);
```

Comment:
I was thinking the same, but there is the same difference between `remove(Object)` and `remove(int)`. This pattern fits better the current behaviour in the generic Collection API.